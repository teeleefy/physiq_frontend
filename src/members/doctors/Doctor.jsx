import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
// import '../styles/Allergy.css'


function Doctor({doctor, memberId}){

    return(
        <>
            <ListGroupItem>
            <Card className="Doctor-card">
                <CardBody>
                    <CardText>
                        <p><b>Doctor Name:</b> <NavLink className="Doctor-name" to={`/member/${memberId}/doctors/${doctor.id}`} key={doctor.id}>
                            {doctor.name}
                        </NavLink></p>
                        <p><b>Specialty:</b> {doctor.specialty}</p>
                        <p><b>Clinic:</b> {doctor.clinic}</p>
                        <p><b>Address:</b> {doctor.address}</p>
                        <p><b>Phone:</b> {doctor.phone}</p>
                        <p><b>Notes:</b> {doctor.notes}</p>
                    </CardText>
                </CardBody>
            </Card>
            </ListGroupItem>
        </>
     )
}

export default Doctor;