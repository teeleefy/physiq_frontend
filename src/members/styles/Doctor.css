.Doctor-card {
    height: 30vh;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Doctor-label {
    font-size: 1.5rem;
}

.Doctor-h1 {
    font-size: 3rem;
    font-weight: 600;
}

.Doctors-navlink {
    text-decoration: none;
    color: white;
}