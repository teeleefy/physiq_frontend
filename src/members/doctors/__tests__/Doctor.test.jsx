import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import Doctor from '../Doctor'

let doctor={name: "Dr. Jones", specialty: "Pulmonology", clinic: "Main Street Pulmonology", address: "100 Main Street", phone: "(555) 555-5555", notes: "See him for asthma management."}
let memberId=1000;

describe('Doctor', () => {
  it('renders the Doctor component', () => {
    renderWithRouter(<Doctor doctor={doctor} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Doctor component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Doctor doctor={doctor} memberId={memberId}/>);
  expect(asFragment()).toMatchSnapshot();
});