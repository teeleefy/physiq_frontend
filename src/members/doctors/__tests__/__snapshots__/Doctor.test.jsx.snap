// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Doctor-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Doctor Name:
          </b>
           
          <a
            class="Doctor-name"
            href="/member/1000/doctors/undefined"
          >
            Dr. Jones
          </a>
        </p>
        <p>
          <b>
            Specialty:
          </b>
           Pulmonology
        </p>
        <p>
          <b>
            Clinic:
          </b>
           Main Street Pulmonology
        </p>
        <p>
          <b>
            Address:
          </b>
           100 Main Street
        </p>
        <p>
          <b>
            Phone:
          </b>
           (555) 555-5555
        </p>
        <p>
          <b>
            Notes:
          </b>
           See him for asthma management.
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
